HyperVExtension/test/HyperVExtension/HyperVExtensionTests/Services/HyperVExtensionIntegrationTest.cs

```diff
@@ -361,4 +378,118 @@ public void TestDevSetupAgentDeployment()
         Assert.IsNotNull(psObject);
         Assert.AreEqual(psObject!.Properties["Status"].Value, "Running");
     }
+
+    /// <summary>
+    /// This test will attempt to create a new VM on the computer using the VMGalleryService.
+    /// It should be ran normally, not in a CI/CD pipeline until we have pipeline machines
+    /// that have Hyper-V enabled and running.
+    /// </summary>
+    [TestMethod]
+    public async Task TestVirtualMachineCreationFromVmGallery()
+    {
+        var vmGalleryService = TestHost!.GetService<IVMGalleryService>();
+        var hyperVProvider = TestHost!.GetService<IComputeSystemProvider>();
+        var expectedVMName = "New Windows 11 VM for Integration test";
+        var imageList = await vmGalleryService.GetGalleryImagesAsync();
+        var smallestImageIndex = await GetIndexOfImageWithSmallestRequiredSpace(imageList);
+        var inputJson = JsonSerializer.Serialize(new VMGalleryCreationUserInput()
+        {
+            NewVirtualMachineName = expectedVMName,
+
+            // Get Image with the smallest size from gallery, we'll use it to create a VM.
+            SelectedImageListIndex = smallestImageIndex,
+        });
+
+        var createComputeSystemOperation = hyperVProvider.CreateCreateComputeSystemOperation(null, inputJson);
+        createComputeSystemOperation!.Progress += OnProgressReceived;
+
+        // Act
+        var createComputeSystemResult = await createComputeSystemOperation!.StartAsync();
+        createComputeSystemOperation!.Progress -= OnProgressReceived;
+
+        // Assert
+        Assert.AreEqual(ProviderOperationStatus.Success, createComputeSystemResult.Result.Status);
+        Assert.AreEqual(expectedVMName, createComputeSystemResult.ComputeSystem.DisplayName);
+        CleanUpVirtualMachine(createComputeSystemResult.ComputeSystem, imageList.Images[smallestImageIndex], expectedVMName);
+    }
+
+    public void CleanUpVirtualMachine(IComputeSystem computeSystem, VMGalleryImage image, string virtualMachineName)
+    {
+        try
+        {
+            var hyperVManager = TestHost!.GetService<IHyperVManager>();
+            var virtualMachine = computeSystem as HyperVVirtualMachine;
+            var hardDisk = virtualMachine!.GetHardDrives().First();
+
+            // remove extracted VHD

```

Comment:
Consider try/catch around individual delete operations to continue cleanup if one of them failed.

Reply:
updated thanks